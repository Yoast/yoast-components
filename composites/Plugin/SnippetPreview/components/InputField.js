import PropTypes from "prop-types";
import React from "react";
import { EditorState, Editor, Modifier } from "draft-js";
import colors from "../../../style-guide/colors.json";
import styled from "styled-components";

const InputFieldContainer = styled.div`
	cursor: text;
	height: 16px;
	border: 1px solid ${ colors.$color_grey };
	width: 100%;
	padding: 3px 5px;
	margin: 5px 1px 1px 1px;
	background-color: ${ colors.$color_white };
	overflow: hidden;
	font-size: 14px;
	font-style: inherit;
	.public-DraftStyleDefault-block {
        white-space: nowrap;
        word-wrap: normal;
    }
`;

export default class InputField extends React.Component {
	constructor( props ) {
		super( props );
		this.state = { editorState: EditorState.createEmpty() };
	}

	render() {
		return (
			<InputFieldContainer onClick={ this.focus.bind( this ) }>
				<Editor
					editorState={ this.state.editorState }
					handlePastedText= { this.handlePastedText.bind( this ) }
					onChange={ this.onChange.bind( this ) }
					placeholder={ this.props.placeholder }
					ref="editor"
					handleReturn={ () => "handled" }
				/>
			</InputFieldContainer>
		);
	}

	onChange( editorState ) {
		this.setState( { editorState } );
	}

	focus() {
		this.refs.editor.focus();
	}

	handlePastedText( text ) {
		this.onChange( EditorState.push(
			this.state.editorState,
			Modifier.replaceText(
				this.state.editorState.getCurrentContent(),
				this.state.editorState.getSelection(),
				text.replace( /\n|\r/g, " " )
			)
		) );
		return "handled";
	}
}

InputField.propTypes = {
	placeholder: PropTypes.string,
};

InputField.defaultProps = {
	placeholder: "",
};
