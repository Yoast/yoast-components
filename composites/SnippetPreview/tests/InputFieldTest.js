import React from "react";
import renderer from "react-test-renderer";
import InputField from "../components/InputField";

jest.mock( "draft-js/lib/generateRandomKey", () => () => "test" );

test( "the input field matches the snapshot", () => {
	const component = renderer.create(
		<InputField placeholder="placeholder"/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
