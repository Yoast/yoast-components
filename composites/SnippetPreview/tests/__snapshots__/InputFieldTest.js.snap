// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the input field matches the snapshot 1`] = `
<div
  className="sc-bdVaJa jIBrAg"
  onClick={[Function]}
>
  <div
    className="DraftEditor-root"
  >
    <div
      className="public-DraftEditorPlaceholder-root"
    >
      <div
        className="public-DraftEditorPlaceholder-inner"
        id="placeholder-test"
      >
        placeholder
      </div>
    </div>
    <div
      className="DraftEditor-editorContainer"
    >
      <div
        aria-activedescendant={undefined}
        aria-autocomplete={undefined}
        aria-describedby="placeholder-test"
        aria-expanded={undefined}
        aria-haspopup={undefined}
        aria-label={undefined}
        aria-owns={undefined}
        autoCapitalize={undefined}
        autoComplete={undefined}
        autoCorrect={undefined}
        className="notranslate public-DraftEditor-content"
        contentEditable={true}
        data-testid={undefined}
        onBeforeInput={[Function]}
        onBlur={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onCopy={[Function]}
        onCut={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseUp={[Function]}
        onPaste={[Function]}
        onSelect={[Function]}
        role="textbox"
        spellCheck={false}
        style={
          Object {
            "outline": "none",
            "whiteSpace": "pre-wrap",
            "wordWrap": "break-word",
          }
        }
        suppressContentEditableWarning={true}
        tabIndex={undefined}
      >
        <div
          data-contents="true"
        >
          <div
            className=""
            data-block={true}
            data-editor="test"
            data-offset-key="test-0-0"
          >
            <div
              className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
              data-offset-key="test-0-0"
            >
              <span
                data-offset-key="test-0-0"
                style={Object {}}
              >
                <br
                  data-text="true"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
